src/main/java/uk/gov/hmcts/darts/common/entity/listener/UserAuditListener.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.darts.common.entity.listener;
+
+import jakarta.persistence.PrePersist;
+import jakarta.persistence.PreUpdate;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import uk.gov.hmcts.darts.authorisation.component.UserIdentity;
+import uk.gov.hmcts.darts.common.entity.UserAccountEntity;
+import uk.gov.hmcts.darts.common.entity.base.CreatedBy;
+import uk.gov.hmcts.darts.common.entity.base.LastModifiedBy;
+
+import java.time.Clock;
+import java.time.OffsetDateTime;
+import java.util.Optional;
+
+@Slf4j
+@NoArgsConstructor
+@AllArgsConstructor
+public class UserAuditListener {
+
+    @Lazy
+    @Autowired
+    private UserIdentity userIdentity;
+
+    @Autowired
+    private Clock clock;
```

Comment:
This wont work here as UserIdentity needs to be lazy loaded to avoid circular references,
However, I have manually created the constructed with the relevant annotations to resolve this